.github/workflows/vm-kernel.yaml

```
@@ -185,14 +190,15 @@ jobs:
         with:
           build-args: KERNEL_VERSION=${{ steps.get-kernel-version.outputs.VM_KERNEL_VERSION }}
           context: neonvm-kernel
-          platforms: linux/amd64
+          platforms: ${{format('linux/{0}', inputs.arch)}}
           # neonvm-kernel/Dockerfile.kernel-builder has different targets for different architectures
           # so we need to specify the target explicitly
-          target: kernel_amd64
+          target: ${{format('kernel_{0}', inputs.arch)}}
           # Push kernel image only for scheduled builds or if workflow_dispatch/workflow_call input is true
           push: true
           pull: true
           file: neonvm-kernel/Dockerfile.kernel-builder
-          cache-from: type=registry,ref=cache.neon.build/vm-kernel:cache
-          cache-to: ${{ github.ref_name == 'main' && 'type=registry,ref=cache.neon.build/vm-kernel:cache,mode=max' || '' }}
+          cache-from: ${{format('type=registry,ref=cache.neon.build/vm-kernel:{0}cache', inputs.arch)}}
+          # Cache the kernel image for future builds, temporary from all branches
+          cache-to: ${{ format('type=registry,ref=cache.neon.build/vm-kernel:{0}cache,mode=max', inputs.arch) || '' }}
```

We usually add "arch" part to image name itself:

```suggestion
          cache-from: ${{format('type=registry,ref=cache.neon.build/vm-kernel-{0}:cache', inputs.arch)}}
          # Cache the kernel image for future builds, temporary from all branches
          cache-to: ${{ format('type=registry,ref=cache.neon.build/vm-kernel-{0}:cache,mode=max', inputs.arch) || '' }}
```